cli/src/config-schema.json

```diff
@@ -332,6 +332,14 @@
                             "type": "string"
                         }
                     },
+                    "diff-mode": {
```

Comment:
Do people have an opinion on `diff-mode` vs `diff-dir-mode`? I prefer the latter, in case there are other kind of options that could be named "mode" are added in the future, but maybe people have better ideas or are fine with `diff-mode`?

We can also rename it in a later PR (especially if this is merged after the release).

Reply:
I don't like `diff-dir-mode` because that is only one of they types.  `diff-dir-more=file-by-file` seems poorly named to me.

Reply:
Yeah, in my mind this means "the approach we take to the fact that the diff contains a whole dir is that we send the dirs to the tool one file at a time".

I'd like a better name, but I don't think this is blocking; we can change it later (we'll have a month to do it without a breaking change if we wait until Wednesday to merge).

Reply:
`diff-dir-singly`? `diff-each-file-in-dir`?

Reply:
"diff-(tool-)exec-mode" or "diff-(tool-)invocation-mode"? ("tool" is omitted because it's under the "merge-tools" section.)

Reply:
went with diff-invocation-mode